### ISSUE TITLE
BP.Table scroll events are processed after table gets unmounted

### ISSUE THREAD

**Issue by N1kto** [palantir/blueprint#4472]
#### Environment

- __Package version(s)__: @blueprintjs/table@3.8.18 (and 3.8.13)
- __Operating System__: Linux
- __Browser name and version__: Chromium 87, Firefox 86

#### Code Sandbox

Link to a minimal repro (fork [this code sandbox](https://codesandbox.io/s/blueprint-sandbox-et9xy)): https://codesandbox.io/s/blueprint-sandbox-forked-iuzkh
#### Steps to reproduce

1. Go to table
2. Scroll the table back and forth
3. Click the "go back" button quickly after the scroll (in real environment, clicking the back button to navigate to the previous location works the same)

#### Actual behavior

Exception is thrown: TypeError: Cannot read property 'offsetWidth' of null

https://user-images.githubusercontent.com/5045863/103891341-f3093100-50e9-11eb-9995-8d4a77be889c.mp4

#### Expected behavior

No exception

#### Possible solution

The scroll event handler should check if the table is still mounted

**Comment by N1kto** [comment]
For those still suffering because of this issue I've come up with a workaround.
1. You need a BP.Table reference
```
import { Table } from  '@blueprintjs/table';
import { useRef } from ''react';
const SomeFancyTable = (fancyTableProps) => {
  const tableRef = useRef<null | Table>(null);
  ...
  return (
    <Table ref={rableRef} {...fancyTableProps} />
  );
}
```
2. Add an unmount handler to your component where you'd manually clear `debouncedViewSyncInterval`. Clearing this interval fixes the issue:
```
useEffect(() => {
      // ensure any scheduled handlers are canceled on unmount
      return () => {
        if (tableRef.current) {
          // dispite quadrantStackInstance being a private member of quadrantStackInstance we still can access it
          const tableQuadrantStackInstance = (tableRef.current as any).quadrantStackInstance;
          clearInterval(tableQuadrantStackInstance?.debouncedViewSyncInterval);
          // also added these 2 `clearTimeouts` calls below (they don't help with original issue though), as it still feels right to call them on unmount
          tableQuadrantStackInstance?.clearTimeouts();
          tableRef.current.clearTimeouts();
        }
      };
    }, []);
```

**Comment by adidahiya** [comment]
@N1kto we have some share component infra for dealing with canceling timeouts on unmount. I've gone ahead and used that here in https://github.com/palantir/blueprint/pull/5124. That should fix the issue, but beware that it will break your workaround because of a private member rename.

**Comment by adidahiya** [comment]
Fixed in v4 by #5124. If there's a strong need for a backport to v3, I'll consider it, but for now I'm trying to only push new code to v4.
